packages/terraform-modules-public/aws/example/variables.tf

```diff
@@ -0,0 +1,215 @@
+}
+
+variable cidr_blocks {
+    type            = object({
+      vpc                 = string
+      subnet_az1_public   = string
+      subnet_az1_private  = string
+      subnet_az2_public   = string
+      subnet_az2_private  = string
+      allowed_ssh_clients = string
+    })
+    description     = "The cidr_blocks for the different subnets in a redundant Celo network"
+    default         = {
+      vpc                 = "10.10.0.0/16"
+      subnet_az1_public   = "10.10.0.0/24"
+      subnet_az1_private  = "10.10.1.0/24"
+      subnet_az2_public   = "10.10.10.0/24"
+      subnet_az2_private  = "10.10.11.0/24"
+      allowed_ssh_clients = "0.0.0.0/0"
+    }
+}
+
+variable key_pair_name {
+    type            = string
+    description     = "SSH key pair name"
+}
+
+variable celo_image {
+  type              = string
+  description       = "Docker image for Celo nodes"
+  default           = "us.gcr.io/celo-testnet/celo-node:1.8"
+}
+
+variable celo_network_id {
+  type              = string
+  description       = "ID of the Celo network to join"
+  default           = "200110"
+}
+
+variable proxies {
+  description = "Configuration for zero or more proxies in each availability zone."
+  type        = object({
+    az1   = map(object({
+      validator_name              = string
+      validator_signer_address    = string
+    }))
+    az2   = map(object({
+      validator_name              = string
+      validator_signer_address    = string
+    }))
+  })
+  default = {
+    az1     = {}
+    az2     = {}
+  }
+  # Here is an example configuration. We recommend putting this into a secret.auto.tfvars file.
+  # default = {
+  #   az1 = {
+  #     myvalidator_az1_01 = {
+  #       validator_name            = "myvalidator_az1_01"
+  #       validator_signer_address  = "0000000011111111222222223333333344444444"
+  #     }
+  #     myvalidator_az1_02 = {
+  #       validator_name            = "myvalidator_az1_02"
+  #       validator_signer_address  = "5555555566666666777777778888888899999999"
+  #     }
+  #   }
+  #   az2 = {
+  #     myvalidator_az2_01 = {
+  #       validator_name            = "myvalidator_az2_01"
+  #       validator_signer_address  = "4444444433333333222222221111111100000000"
+  #     }
+  #     myvalidator_az2_02 = {
+  #       validator_name            = "myvalidator_az2_02"
+  #       validator_signer_address  = "9999999988888888777777776666666655555555"
+  #     }
+  #   }
+  #
+  # }
+}
+
+variable validators {
+    description     = "Configuration for zero or more validators in each availability zone"
+    type            = object({ 
+      az1 = map(object({
+        name                          = string
+        signer_address                = string
+        signer_private_key            = string
+        signer_private_key_password   = string
+        signer_private_key_filename   = string
+        proxy_enode                   = string
+        proxy_private_ip              = string
```

Comment:
I thought about this and ideally would feed in data from the proxy dynamically in the way you mentioned. Unfortunately, the enode makes this tricky. I'd need a way to get the enode from the proxy which isn't trivial at provisioning time.

So to provision a validator, the user needs to provision the proxy, harvest the values, and then put them in their variables. I think a more automated solution would be a candidate for a future enhancement, so not planning to make a change here.

Reply:
Maybe I am missing something, but the enode is the public key. So if the user provides as an input the proxy's nodekey (private part), the enode would be also known in advance.

Reply:
So how do you deploy it the way it is? Do you target only proxies first? I still think this would be good to have, and we could explicitly set the proxy private key. The enode is the public key minus the first two characters, docs are being written up for this too. I think this is how we do the GCP and azure infra right? Wdyt @jcortejoso ?

Reply:
@tkporter Yes exactly what I mean. As for [GCP](https://github.com/celo-org/celo-monorepo/blob/master/packages/terraform-modules-public/testnet/variables.tf#L114) and [Azure](https://github.com/celo-org/celo-monorepo/blob/master/packages/azure/template.json#L1038) modules, if the user provides a private key for the proxy nodekey, the enode will be explicitly known and will be the uncompressed public key minus the initial `04` characters. It is explained here https://github.com/celo-org/celo-monorepo/blob/master/packages/terraform-modules-public/example/README.md#ethereum-accounts

Reply:
Ah, very interesting. Thanks for this - I didn't know enodes were deterministic in this way. Today, this approach requires you to stand up the proxy and then use params from the proxy in the validator, so you have to run terraform apply multiple times to stand up the cluster (not ideal).

I will update this to allow it all to stand up in one apply call.

Reply:
(The original implementation was modeled after the Running a Validator docs, which made me assume a proxy needed to be running before a validator config could be known)